src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs

```diff
@@ -796,7 +796,7 @@ void SearchButtonClicked(object sender, EventArgs e)
 
 		void SetSearchBarIcon(UISearchBar searchBar, ImageSource source, UISearchBarIcon icon)
 		{
-			source.LoadImage(source.FindMauiContext(), image =>
+			source.LoadImage(MauiContext, image =>
```

Comment:
If MauiContext is null at this point, this would crash.

FindMauiContext if cannot find the parent can fallback to the App MauiContext: https://github.com/dotnet/maui/blob/87f8b927b0bc48cbcb170a864d8d4b2a0540d2e6/src/Controls/src/Core/ViewExtensions.cs#L318

So, with the changes, before return I would try to create something like:

```
var mauiContext = MauiContext  ?? Application.Current?.FindMauiContext();

if (mauiContext is null)
	return;
```

Or, use:

`source.FindMauiContext(true)`

Reply:
@jsuarezruiz, Thanks for the suggestion. I have updated it accordingly.